### PR TITLE
Add docstrings

### LINKED ISSUE #42: Missing docstring for annotation

### LINKED ISSUE #43: Missing docstring for axis & decomposition macros

### PR DESCRIPTION
**Pull request by guimarqu**
fix #42, fix #43

### CODE REVIEW

File: src/formulations.jl [hunk 1]
```diff
@@ -64,6 +80,27 @@ function _specify!(sp::SubproblemForm, lm::Real, um::Real, oracle::Union{Nothing
...
+        upper_multiplicity = 1,
+        solver = nothing
+    )
+
+Method that allows the user to specify additional property of the subproblems.
+
+The multiplicity of `subproblem` is the number of times that the same independant
+block shaped by the subproblem in the coefficient matrix appears in the model.
+It is also the number of solutions to the subproblem can appear in the solution 
+of the original problem.
```
Comment (guimarqu) [inline src/formulations.jl:96]:
```suggestion
It is equivalent to the number of solutions to the subproblem that can appear in the solution 
of the original problem.
```

File: src/BlockDecomposition.jl [hunk 2]
```diff
@@ -51,6 +51,17 @@ function optimize!(m::JuMP.Model)
...
+"""
+    annotation(node)
+
+Return the annotation that describes the master/subproblem of a given node of
+the decomposition tree.
+
+    annotation(model, variable)
+    annotation(model, constraint)
+
+Return the subproblem membership of a variable or a constraint.
```
Comment (guimarqu) [inline src/BlockDecomposition.jl:63]:
```suggestion
Return the subproblem to which a variable or a constraint belongs.
```

File: src/formulations.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (guimarqu) [reply src/formulations.jl:96]: We can say "it is equivalent to"

File: src/BlockDecomposition.jl (hunk 2 above)
[the remark this answers is not in the document]
Comment (guimarqu) [reply src/BlockDecomposition.jl]: Yes it is.

**Review by laradicp — APPROVED** [review]

**Pull request merged**